pandas/core/indexes/base.py

```diff
@@ -1038,6 +1039,31 @@ def _to_embed(self, keep_tz=False, dtype=None):
 
         return self.values.copy()
 
+    def _as_best_array(self):
```

Comment:
I found a need for a method like this. It may be good to add to Series as well, and maybe make public.

Essentially, we want a clear way to get the "best" `.values`. For Indexes with ExtensionArray types (`Categorical` for now, `Interval`, `Period` soon) that's the extension array. Otherwise, it's an ndarray (or I think `DatetimeIndex for datetime w/ tz right now).

Reply:
(this will also depend on decision if we will change return value for `.values` for series with period and interval objects) 
For Series we already have something like `._values`. Won't that be the equivalent there?

Reply:
Yes, I think so. Maybe I should re-use that convention.

Unfortunately, `DatetimeIndex[tz]._values` returns an `ndarray`, w/o the TZ, where it "should" return a `DatetimeIndex` (and eventually an ExtensionArray). I'll see how badly changing that breaks things.

Reply:
yes ths should be equiv of ``_values``. DTI should be fixed as a pre-cursor PR. no problem making a different method name for this, but then simply rename _values to that name (I don't personally like _as_best_array), maybe (_as_array) is better. Should again be a pre-cursor independent PR.